lib/util/git/auth.ts

```diff
@@ -1,10 +1,30 @@
+import type { PlatformId } from '../../constants/platforms';
 import { logger } from '../../logger';
 import type { HostRule } from '../../types';
 import { detectPlatform } from '../common';
+import { find, getAll } from '../host-rules';
 import { regEx } from '../regex';
+import { createURLFromHostOrURL, validateUrl } from '../url';
 import type { AuthenticationRule } from './types';
 import { parseGitUrl } from './url';
 
+const githubApiUrls = new Set([
+  'github.com',
+  'api.github.com',
+  'https://api.github.com',
+  'https://api.github.com/',
+]);
+
+const standardGitAllowedHostTypes = new Set<string>([
+  // All known git platforms
+  'azure',
+  'bitbucket',
+  'bitbucket-server',
+  'gitea',
+  'github',
+  'gitlab',
+] satisfies PlatformId[]);
```

Comment:
```suggestion
const standardGitAllowedHostTypes = [
  // All known git platforms
  'azure',
  'bitbucket',
  'bitbucket-server',
  'gitea',
  'github',
  'gitlab',
] satisfies PlatformId[];
```

the set is only overhead here.